webhook.go

```diff
@@ -272,6 +283,10 @@ func updateTCPListener(listener *v1.Listener) {
 
 // clusters handles the CDS hook and inserts the dikastes cluster
 func clusters(req *restful.Request, resp *restful.Response) {
+	// TODO(saumoh): when we don't change the cluster body (configOptions.SetCluster == false)
+	// we should just do a io.Copy(resp, req.Request.Body)
+	// but that results in Envoy rejecting the configuration.
+	// Hence, read, deconstruct, no-op, write to output!
```

Comment:
Wait, what? That smells really bad. What error does Envoy give?

Reply:
The exact error is:
`[2018-03-15 22:15:56.687][74][warning][upstream] external/envoy/source/common/upstream/cds_subscription.cc:67] cds: fetch failure: JSON supplied is not valid. Error(offset 3852, line 114): Missing a closing quotation mark in string.
`
It does not clearly say which line or anything but I can throw in some more debugs to see what's going on in Pilot.
Additionally I agree for v1 we should just drop modifying the cluster. Let me open a bug (internally) so that I can track this specific issue.
Will send another patch with the new change set.

Reply:
Interesting. I guess the right thing to do is `Read()` to a buffer then `Write()` to the response. No need to decode & re-encode.

If you don't mind, could you fix up the other endpoints that do this?

Reply:
yes. I believe that is what we have to do. I've fixed other pass through as well. ptal. thx